src/app/shared/components/zooming-image-viewer/zooming-image-viewer.component.ts

```diff
@@ -0,0 +1,100 @@
+/* @format */
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  OnDestroy,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { RecordVO } from '@models/index';
+import * as OpenSeaDragon from 'openseadragon';
+import { ZoomEvent, FullScreenEvent } from 'openseadragon';
+
+@Component({
+  selector: 'pr-zooming-image-viewer',
+  templateUrl: './zooming-image-viewer.component.html',
+  styleUrl: './zooming-image-viewer.component.scss',
+})
+export class ZoomingImageViewerComponent implements AfterViewInit, OnDestroy {
+  @Input() public item: RecordVO;
+
+  @Output() public disableSwipe = new EventEmitter<boolean>(false);
+  @Output() public isFullScreen = new EventEmitter<boolean>(false);
+
+  public viewer: OpenSeaDragon.Viewer;
+
+  private initialZoom: number;
+
+  @ViewChild('viewer') element!: ElementRef<HTMLElement>;
+
+  constructor(
+    @Inject('openseadragon') private imageViewerFn: typeof OpenSeaDragon,
```

Comment:
This follows a similar pattern to mocking the `Image` type earlier. For some reason, OpenSeaDragon takes a long time to initialize in unit tests (to the point where it was reliably causing test failures since the tests were timing out!). I traced the performance issues specifically to the `OpenSeaDragon()` initialization call, so the best course of action was just to mock the OpenSeaDragon object. This is unfortunate because although we generally don't want to test 3rd party libraries, OpenSeaDragon gives us the `raiseEvent` method that would have allowed us to cover all of our custom event handlers easily without any mocking... if only it weren't so slow!

<details>
<summary>Additional mysterious details which are not relevant anymore</summary>
For some reason changing image URLs seemed to have a noticeably significant impact on test run times. Choosing a valid image URL would make tests time out every time, while choosing an invalid one would allow them to at least pass within the 5000ms window most of the time. This seems to indicate that somehow OpenSeaDragon is fetching and initializing images as part of its setup process. But the call to initialize OpenSeaDragon is not in the form of an async/await/promise call. And the tests themselves also do not have any async behavior besides the shallow rendering. The only way to load image assets in JavaScript is through asynchronous calls though, so somehow in unit tests OpenSeaDragon was causing blocking issues with async calls. I don't even know how that is remotely possible. If event handlers weren't called until the images were loaded, this would just cause the tests to fail instantly since we don't have any asynchronous expects in the tests. My only theory is that there was some kind of weird interaction between OpenSeaDragon and shallow-render? I still don't know how changing image URLs would affect that though. This issue gave me a headache.</details>